orbit/cmd/desktop/desktop.go

```diff
@@ -187,6 +190,48 @@ func main() {
 			migrateMDMItem.Hide()
 		}
 
+		// TODO: we can probably extract this into a function that sets up both the migrator and the
+		// offline watcher
+		if runtime.GOOS == "darwin" {
+
+			dir, err := migration.Dir()
+			if err != nil {
+				log.Fatal().Err(err).Msg("getting directory for MDM migration file")
+			}
+
+			mrw := migration.NewReadWriter(dir, constant.MigrationFileName)
+
+			// swiftDialogCh is a channel shared by the migrator and the offline watcher to
+			// coordinate the display of the dialog and ensure only one dialog is shown at a time.
+			// we use channel buffer size of 1 to allow one dialog at a time with non-blocking sends.
+			swiftDialogCh := make(chan struct{}, 1) // TODO: when should we close this channel?
+
+			_, swiftDialogPath, _ := update.LocalTargetPaths(
+				tufUpdateRoot,
+				"swiftDialog",
+				update.SwiftDialogMacOSTarget,
+			)
+			mdmMigrator = useraction.NewMDMMigrator(
+				swiftDialogPath,
+				15*time.Minute,
+				&mdmMigrationHandler{
+					client:      client,
+					tokenReader: &tokenReader,
+				},
+				mrw,
+				swiftDialogCh,
+			)
+
+			useraction.StartMDMMigrationOfflineWatcher(ctx, client, swiftDialogPath, swiftDialogCh, migration.FileWatcher(mrw))
```

Comment:
would it be possible to start the watcher only if migration is enabled?

Reply:
I thought about that and opted for simplicity in the interest of time and to save us from having to work through a bunch of edge cases. I doesn't seem like the loop adds much overhead since it is pretty simple. If the file isn't present, it doesn't even try to ping the server.

Reply:
I'm worried about the logs and people complaining about it to be honest

Reply:
We can definitely trim down the logs. I think most of the logs are at the debug level already.

Reply:
okay, if there's not much logging, and we're careful when we open swiftDialog sounds good!